common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -573,7 +612,8 @@ private static int collationNameToId(
           }
         }
         if (lastPos == -1) {
-          throw collationInvalidNameException(originalName);
+          throw collationInvalidNameException(
+            originalName, ICULocaleNames, List.of("_CI", "_AI", "_CS", "_AS"));
```

Comment:
You can use `collationName` instead of `originalCollationName` and avoid uppercasing in exception utility function.

Reply:
I think that we want to keep originalName because that is what customer gave to us and we want to report that as the error. E.g. if I were to put eN_US I would like to get "invalid collation name eN_US, suggestion en_USA".
If I were to provide collationName here the message would be "invalid collation name en_US (not eN_US).